core/src/main/scala/com/databricks/labs/remorph/parsers/intermediate/functions.scala

```diff
@@ -12,299 +12,329 @@ case class CallFunction(function_name: String, arguments: Seq[Expression]) exten
+    case CaseInsensitiveCallFunction("UNIX_TIMESTAMP", args) => UnixTimestamp(args.head, args(1))
+    case CaseInsensitiveCallFunction("UUID", _) => Uuid()
+    case CaseInsensitiveCallFunction("VAR_POP", args) => VariancePop(args.head)
+    case CaseInsensitiveCallFunction("VAR_SAMP", args) => VarianceSamp(args.head)
+    case CaseInsensitiveCallFunction("VERSION", _) => SparkVersion()
+    case CaseInsensitiveCallFunction("WEEKDAY", args) => WeekDay(args.head)
+    case CaseInsensitiveCallFunction("WEEKOFYEAR", args) => WeekOfYear(args.head)
+    case CaseInsensitiveCallFunction("WHEN", _) =>
       throw new IllegalArgumentException("WHEN (CaseWhen) should be handled separately")
-    case CallFunction("WIDTH_BUCKET", args) => WidthBucket(args.head, args(1), args(2), args(3))
-    case CallFunction("WINDOW", _) =>
+    case CaseInsensitiveCallFunction("WIDTH_BUCKET", args) => WidthBucket(args.head, args(1), args(2), args(3))
+    case CaseInsensitiveCallFunction("WINDOW", _) =>
       throw new IllegalArgumentException("WINDOW (TimeWindow) should be handled separately")
-    case CallFunction("XPATH", args) => XPathList(args.head, args(1))
-    case CallFunction("XPATH_BOOLEAN", args) => XPathBoolean(args.head, args(1))
-    case CallFunction("XPATH_DOUBLE", args) => XPathDouble(args.head, args(1))
-    case CallFunction("XPATH_FLOAT", args) => XPathFloat(args.head, args(1))
-    case CallFunction("XPATH_INT", args) => XPathInt(args.head, args(1))
-    case CallFunction("XPATH_LONG", args) => XPathLong(args.head, args(1))
-    case CallFunction("XPATH_SHORT", args) => XPathShort(args.head, args(1))
-    case CallFunction("XPATH_STRING", args) => XPathString(args.head, args(1))
-    case CallFunction("XXHASH64", args) => XxHash64(args)
-    case CallFunction("YEAR", args) => Year(args.head)
-    case CallFunction("ZIP_WITH", args) => ZipWith(args.head, args(1), args(2))
+    case CaseInsensitiveCallFunction("XPATH", args) => XPathList(args.head, args(1))
+    case CaseInsensitiveCallFunction("XPATH_BOOLEAN", args) => XPathBoolean(args.head, args(1))
+    case CaseInsensitiveCallFunction("XPATH_DOUBLE", args) => XPathDouble(args.head, args(1))
+    case CaseInsensitiveCallFunction("XPATH_FLOAT", args) => XPathFloat(args.head, args(1))
+    case CaseInsensitiveCallFunction("XPATH_INT", args) => XPathInt(args.head, args(1))
+    case CaseInsensitiveCallFunction("XPATH_LONG", args) => XPathLong(args.head, args(1))
+    case CaseInsensitiveCallFunction("XPATH_SHORT", args) => XPathShort(args.head, args(1))
+    case CaseInsensitiveCallFunction("XPATH_STRING", args) => XPathString(args.head, args(1))
+    case CaseInsensitiveCallFunction("XXHASH64", args) => XxHash64(args)
+    case CaseInsensitiveCallFunction("YEAR", args) => Year(args.head)
+    case CaseInsensitiveCallFunction("ZIP_WITH", args) => ZipWith(args.head, args(1), args(2))
     case _ => call // fallback
   }
 }
 
+class SnowflakeCallMapper extends CallMapper {
```

Comment:
does not belong to this file. add it to `rules` package under `showflake` and create a dedicated optimizer rule